### ISSUE TITLE
Noisy train loss

### ISSUE THREAD

**Issue by muchas** [snap-stanford/pretrain-gnns#2]
Hi there,

Great work, thank you for cool insights into graph models pre-training!

I've tried to reproduce the results, but when running GIN pretraining in masking + supervised setup I encountered a bit unstable train loss.

<img width="410" alt="GIN - pretraining GNNS supervised" src="https://user-images.githubusercontent.com/932227/75322613-64921500-5873-11ea-9695-0eaa8d299262.png">

Is is inherit to chembl multi-task setup or should I expect it to converge?

**Comment by weihua916** [comment]
Hi, thanks for your interest!
It seems that you have evaluated the loss for each mini-batch, which is inherently noisy. You should be able to see the loss converging if you plot the loss for each epoch.

**Comment by muchas** [comment]
Ah, you were right. When training bigger model (~40M params) the convergence is also more visible. Thanks for the quick answer!
